### ISSUE TITLE
专家间的损失有问题

### ISSUE THREAD

**Issue by mikumifa** [zjukg/MoMoK#13]
strs, imgs, txts这3个embedding的维度为（bzs,n_exp,embed_dim）
但是random.sample(range(self.num), k=2）取的确实第一个维度
如下面的代码，代码位置https://github.com/zjukg/MoMoK/blob/master/models/modules.py#L85
```python
    def forward(self, embeddings):
        strs, imgs, txts = embeddings
        idx1, idx2 = random.sample(range(self.num), k=2)
        str1, str2 = strs[idx1], strs[idx2]
        img1, img2 = imgs[idx1], imgs[idx2]
        txt1, txt2 = txts[idx1], txts[idx2]
        mi_loss = (
            self.str_estimator(str1, str2)
            + self.img_estimator(img1, img2)
            + self.txt_estimator(txt1, txt2)
        ) / 3.0
        return mi_loss

```
这地方是不是错了？

**Comment by mikumifa** [comment]
（而且还有一个问题，
       estimator.eval()之后并不会关闭计算图，可能还需要with torch.no_grad():
所以第一次  loss.backward()时候，还是会更新estimator的计算图，
不过因为分2个优化器了，结果应该不影响

**Comment by Zhang-Each** [comment]
由于服务器上的版本管理的问题，我们开源的代码中存在一些问题，感谢你提交的pull request。
